model/waterfall.go

```diff
@@ -0,0 +1,168 @@
+package model
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	DefaultWaterfallVersionLimit = 5
+	MaxWaterfallVersionLimit     = 300
+)
+
+type WaterfallTask struct {
+	Id          string `bson:"_id" json:"_id"`
+	DisplayName string `bson:"display_name" json:"display_name,omitempty"`
+	Status      string `bson:"status" json:"status,omitempty"`
+}
+
+type WaterfallBuild struct {
+	Id          string          `bson:"_id" json:"_id"`
+	Activated   bool            `bson:"activated" json:"activated,omitempty"`
+	DisplayName string          `bson:"display_name" json:"display_name,omitempty"`
+	Version     string          `bson:"version" json:"version,omitempty"`
+	Tasks       []WaterfallTask `bson:"tasks" json:"tasks,omitempty"`
+}
+
+type WaterfallBuildVariant struct {
+	Id          string           `bson:"_id" json:"_id"`
+	DisplayName string           `bson:"display_name" json:"display_name,omitempty"`
+	Builds      []WaterfallBuild `bson:"builds" json:"builds,omitempty"`
+}
+
+type WaterfallOptions struct {
+	Limit      int
```

Comment:
nit: might help to put `bson:"-" json:"-"` tags on this to be explicit about the fact that these are just options to pass into the query, and not actually used for any returned DB data.